consensus/polybft/consensus_runtime.go

```diff
@@ -400,15 +410,13 @@ func (c *consensusRuntime) restartEpoch(header *types.Header) error {
 // for state sync events starting from given index and saves the message in database
 func (c *consensusRuntime) buildCommitment(epoch, fromIndex uint64) (*Commitment, error) {
 	// if it is not already built in the previous epoch
-	stateSyncEvents, err := c.state.getStateSyncEventsForCommitment(fromIndex, fromIndex+stateSyncMainBundleSize)
-	if err != nil {
-		if errors.Is(err, ErrNotEnoughStateSyncs) {
-			c.logger.Info("[buildCommitment] Not enough state syncs to build a commitment",
-				"epoch", epoch, "from state sync index", fromIndex)
-			// this is a valid case, there is not enough state syncs
-			return nil, nil
-		}
-
+	stateSyncEvents, err := c.state.getStateSyncEventsForCommitment(fromIndex, fromIndex+stateSyncMainBundleSize-1)
+	if len(stateSyncEvents) == 0 || (err != nil && errors.Is(err, errNotEnoughStateSyncs)) {
```

Comment:
I believe we don't log an error here. How do you think does it worth to distinguish this 2 cases with len(stateSyncEvents) == 0 and errors.Is() ?

Reply:
Yes, this should be logged, you are right.

Reply:
Yes, this error should definitely be returned.

Fix: https://github.com/0xPolygon/polygon-edge/pull/785/commits/e0dc1bd33b0667bdac9a1142a52fb84f0fde749f